src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +66,243 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
+float FlightTaskAutoFollowTarget::calculate_gimbal_height(float target_height) {
+	float gimbal_height{0.0f};
+
+	switch (_param_nav_ft_alt_m.get()) {
+		case FOLLOW_ALTITUDE_MODE_TRACK_TERRAIN:
+			// Point the gimbal at the ground level in this tracking mode
+			gimbal_height = _dist_to_ground;
+			break;
+
+		case FOLLOW_ALTITUDE_MODE_TRACK_TARGET:
+			// Point the gimbal at the target's 3D coordinates
+			gimbal_height = -_position(2) - target_height;
+			break;
 
-	if (_param_nav_ft_delc.get() == 0) {
-		_target_position_filter.update(pos_ned_est);
+		case FOLLOW_ALTITUDE_MODE_CONSTANT:
 
-	} else {
-		// Use a predicted target's position to compensate the filter delay to some extent.
-		const Vector3f target_predicted_position = predict_future_pos_ned_est(POSITION_FILTER_ALPHA, pos_ned_est, vel_ned_est,
-				acc_ned_est);
-		_target_position_filter.update(target_predicted_position);
-	}
+		//FALLTHROUGH
 
-	return _target_position_filter.getState();
+		default:
+			gimbal_height = _home_position_z -_position(2); // Assume target is at home position's altitude
+			break;
+	}
 
+	return gimbal_height;
 }
 
+
 bool FlightTaskAutoFollowTarget::update()
 {
 	_follow_target_estimator_sub.update(&_follow_target_estimator);
-	follow_target_status_s follow_target_status{};
+
+	follow_target_status_s follow_target_status{}; // Debugging uORB message for follow target status
 
 	if (_follow_target_estimator.timestamp > 0 && _follow_target_estimator.valid) {
-		const Vector3f pos_ned_est{_follow_target_estimator.pos_est};
-		const Vector3f vel_ned_est{_follow_target_estimator.vel_est};
-		const Vector3f acc_ned_est{_follow_target_estimator.acc_est};
-		const Vector3f target_position_filtered = calculate_target_position_filtered(pos_ned_est, vel_ned_est, acc_ned_est);
-		const Vector2f offset_vector_filtered = calculate_offset_vector_filtered(vel_ned_est);
-		const Vector3f drone_desired_position = calculate_drone_desired_position(target_position_filtered,
-							offset_vector_filtered);
+		// Update second order target pose filter, with a valid data
+		update_target_pose_filter(_follow_target_estimator);
 
-		// Set position and velocity setpoints
-		float desired_velocity_ff_scale = 0.0f;  // Used to ramp up velocity feedforward, avoiding harsh jumps in the setpoints
+		const Vector3f target_position_filtered = _target_pose_filter.getState();
+		const Vector3f target_velocity_filtered = _target_pose_filter.getRate();
+
+		// Calculate offset 2D vector to target
+		_drone_to_target_vector  = Vector2f(target_position_filtered.xy()) - Vector2f(_position.xy());
+		// Calculate heading to the target (for Yaw setpoint)
+		_drone_to_target_heading = atan2f(_drone_to_target_vector(1), _drone_to_target_vector(0));
+		// Calculate current orbit angle around the target, taken into consideration in RC Follow Angle Adjustment
+		_measured_orbit_angle = matrix::wrap_pi(_drone_to_target_heading + M_PI_F);
```

Comment:
It's calculating the atan2f of the target to drone, so it is essentially 'drone to target' + 180 degrees, I have added additional commit explaining this.

I did this since I thought doing atan2f again for just the negative- vector is a waste of computational resource (although it would be more intuitive) : e.g. `atan2f(_target_to_drone_vector(1), _target_to_drone_vector(0))`